mesh/bitquery_sol_token_info_agent.py

```diff
@@ -0,0 +1,493 @@
+import datetime
+import requests
+import os
+from .mesh_agent import MeshAgent, with_cache, with_retry, monitor_execution
+from core.llm import call_llm_async, call_llm_with_tools_async
+from typing import List, Dict, Any
+import json
+
+class BitquerySolanaTokenInfoAgent(MeshAgent):
+    def __init__(self):
+        super().__init__()
+        self.metadata.update({
+            'name': 'Bitquery Solana Token Info Agent',
+            'version': '1.0.0',
+            'author': 'Heurist team',
+            'author_address': '0x7d9d1821d15B9e0b8Ab98A058361233E255E405D',
+            'description': 'Fetches Solana token trading information and trending tokens from Bitquery',
+            'inputs': [
+                {
+                    'name': 'query',
+                    'description': 'Natural language query about a Solana token or a request for trending tokens',
+                    'type': 'str',
+                    'optional': False
+                }
+            ],
+            'outputs': [
+                {
+                    'name': 'response',
+                    'description': 'Natural language explanation of the token trading information or trending tokens',
+                    'type': 'str'
+                },
+                {
+                    'name': 'data',
+                    'description': 'Structured token trading data or trending tokens data',
+                    'type': 'dict'
+                }
+            ],
+            'external_apis': ['bitquery'],
+            'tags': ['DeFi', 'Market Data', 'Solana']
+        })
+
+    def get_system_prompt(self) -> str:
+        return (
+            "You are a helpful assistant that can fetch Solana token trading information from Bitquery and provide trending tokens on the Solana network.\n"
+            "For detailed trading information, provide the Solana token mint address. The data includes:\n"
+            "1. Price information (open, high, low, close)\n"
+            "2. Trading volume\n"
+            "3. Time-series data in 5-minute intervals\n\n"
```

Comment:
The example output contains an incomplete markdown table which might be hard to interpret. Try adding "You should describe the recent hour price data in natural language, including the open, high, low, close price"